R/meta.R

```diff
@@ -0,0 +1,52 @@
+
+.roxygen_meta <- new.env(parent = emptyenv())
+
+roxy_meta_get <- function(name, default = NULL) {
+  if (exists(name, envir = .roxygen_meta))
+    get(name, envir = .roxygen_meta)
+  else
+    NULL
+}
+
+roxy_meta_set <- function(key, value) {
+  assign(key, value, envir = .roxygen_meta)
+}
+
+roxy_meta_load <- function(base_path = getwd()) {
+
+  meta_dir <- file.path(base_path, "man/roxygen")
+  if (!utils::file_test("-d", meta_dir))
+    return(FALSE)
+
+  pattern <- "^meta[.][Rr]"
+  meta_files <- list.files(meta_dir, pattern = pattern, full.names = TRUE)
+  if (length(meta_files) == 0)
+    return(FALSE)
+
+  if (length(meta_files) > 1) {
+    rlang::abort("Multiple 'man/roxygen/meta.R' files found")
+  }
+
+  parsed <- tryCatch(parse(meta_files), error = identity)
+  if (inherits(parsed, "error")) {
+    message <- "Could not parse 'man/roxygen/meta.R'"
+    rlang::abort(message, parent = parsed)
+  }
+
+  # TODO: appropriate evaluation environment?
```

Comment:
`baseenv()` might be more appropriate? So that you have to qualify any package functions?

Reply:
One thought I had: should `meta.R` be able to have access to functions used in your own package (e.g. your helper functions)?

Reply:
My gut feeling is no — these should be fixed options. (And we can always relax in the future; it'd be hard to rollback to only evaluating in base env if we allow package env now)